### PR TITLE
Added ASCII to HEX conversion goodie

### PR DESCRIPTION
**Pull request by Sanguinary**
https://duck.co/ia/view/ascii_to_hex

### CODE REVIEW

**Comment by daxtheduck** [comment]
@Sanguinary Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by tagawa** [comment]
@Sanguinary, thanks again for creating the ASCII-HEX conversion IA! You’ve been doing a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in conversions.

When you're ready to move on, please:

- Create a new Instant Answer for searches you do related to conversions

OR

- Test live conversion Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

File: lib/DDG/Goodie/ASCIIToHEX.pm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+handle remainder => sub {
+    
+    my $original_string = $_;
+    my $result = unpack('H*', $original_string);
+    
+    # Separates the numbers into sets of two
+    $result =~ s/..\K(?=.)/ /sg;
+    
+    return "$original_string converts to $result",
+        structured_answer => {
```
Comment (mintsoft) [inline lib/DDG/Goodie/ASCIIToHEX.pm]: We're trying to move away from this older output format; can you switch to use the text template instead? There's an example here: https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/BashPrimaryExpressions.pm#L100

File: t/ASCIIToHEX.t [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+use Test::More;
+use DDG::Test::Goodie;
+
+zci answer_type => "ascii_to_hex";
+zci is_cached   => 1;
+
+ddg_goodie_test(
+    [qw( DDG::Goodie::ASCIIToHEX )],
+    'ascii to hex 12345678' => test_zci(
+        '12345678 converts to 31 32 33 34 35 36 37 38',
```
Comment (mintsoft) [inline t/ASCIIToHEX.t]: When the tests are converted to account for the new format structured_answer can you create a sub called `generate_test` which wraps the shape of the map?

**Comment by mintsoft** [comment]
@Sanguinary Apologies for the epic delay here; this PR slipped through the cracks

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ASCII HEX conversion)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ascii_to_hex_mobile.png)

**Comment by moollaza** [comment]
@Sanguinary this is now live on beta -- it looks like your example query should be changed (because it shows no result) and you'll need to update the code so that we don't return an empty result for any other queries 👍

File: lib/DDG/Goodie/ASCIIToHEX.pm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+use strict;
+
+zci answer_type => 'ascii_to_hex';
+zci is_cached => 1;
+
+triggers startend => 'ascii to hex', 'convert ascii hex', 'ascii hex conversion', 'convert ascii to hex', 'ascii hex';
+
+handle remainder => sub {
+    
+    my $original_string = $_;
```
Comment (moollaza) [inline lib/DDG/Goodie/ASCIIToHEX.pm]: You should `return unless $_;` to ensure we have a remainder here

File: lib/DDG/Goodie/ASCIIToHEX.pm (hunk 1 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/ASCIIToHEX.pm]: @Sanguinary This is now ***required*** - so we should update to use templates!

File: t/ASCIIToHEX.t (hunk 2 above)
Comment (GuiltyDolphin) [inline t/ASCIIToHEX.t]: @Sanguinary If you have a look in https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/template/t/Example.t you should see a good example of how tests should be laid out.

File: lib/DDG/Goodie/ASCIIToHEX.pm (hunk 3 above)
Comment (mintsoft) [inline lib/DDG/Goodie/ASCIIToHEX.pm]: Good catch!

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/ASCIIToHEX.pm]: Probably `return unless $_ eq ''` seeing as we are working with numbers.

**Comment by mintsoft** [comment]
Hiya @Sanguinary Just checking in, are you still interested in finishing this off?

**Comment by mintsoft** [comment]
@Sanguinary I'm going to mark this as abandoned for now; feel free to reply and I'll reopen it up :)

**Pull request closed without merging**